flink-yarn/src/main/java/org/apache/flink/yarn/configuration/YarnConfigOptions.java

```diff
@@ -344,6 +344,13 @@
                     .withDescription(
                             "A comma-separated list of additional Kerberos-secured Hadoop filesystems Flink is going to access. For example, yarn.security.kerberos.additionalFileSystems=hdfs://namenode2:9002,hdfs://namenode3:9003. The client submitting to YARN needs to have access to these file systems to retrieve the security tokens.");
 
+    public static final ConfigOption<Boolean> YARN_SECURITY_ENABLED =
```

Comment:
I think whether to fetch delegation token and whether "security is enabled" are different things. So I would suggest rename this option to something like `KERBEROS_USE_DELEGATION_TOKEN`.

Besides, delegation token is not specific to yarn. I think spark supports delegation token for both yarn and mesos. So I also suggest move this option to `SecurityOptions` in flink-core, alongside with other kerberos configurations.

Reply:
Yes. Rename is ok. But `KERBEROS_USE_DELEGATION_TOKEN` maybe not suitable.
Actually, this option indicates Flink whether to fetch delegation token actively, not no use delegation token.

Reply:
Hmm, could you please explain in which situation would a user want to fetch delegation tokens but not to use them? Or what a user need to do to really **use** delegation tokens?

Reply:
Hadoop delegation token is used to access to HDFS/HBase/Hive service. So when you want to be interactive with HDFS, you will need it.
Two options can be used to access above services.
First is to fetch delegation token using Keytab.
Second is to get it from current ugi, if current ugi has the  delegation token which is already existing and can be reused.

Refer to: https://blog.cloudera.com/hadoop-delegation-tokens-explained/

Reply:
If this option only controls whether we fetch delegation tokens, then maybe we can call it `KERBEROS_FETCH_DELEGATION_TOKEN `?

Reply:
+1. Got it.

Reply:
> +1. Got it.

After rechecking the delegation token usage scope, i think it's only specific to Hadoop and related framework, like Hive/HBase. Right?
If not, can you give me some doc or issue link on Spark mesos delegation token? Thanks.

@lirui-apache

Reply:
Delegation tokens are usually used for a distributed job to authenticate with a Hadoop-based service like Hive or HBase. But it should be orthogonal to the resource management framework like yarn, mesos, or k8s. Spark [doc](https://spark.apache.org/docs/latest/security.html#kerberos) indicates that delegation tokens are supported for yarn and mesos. And its delegation token implementations (e.g. `HadoopDelegationTokenManager`, `HadoopDelegationTokenProvider`) are in spark-core.

Actually other projects may also implement their own delegation token mechanisms like what [kafka](https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_delegation.html) does. But I guess that exceeds the scope of this ticket.

Reply:
Done. Please review it.